docs/roadmap.md

```diff
@@ -0,0 +1,83 @@
+# Roadmap
+
+This is a list of goals we have. Many of them are quite independent.
+
+> **Note:** Most people contributing to Jujutsu do so in their spare time, which
+>  means that we cannot attach any target dates to any of the goals below.
+
+## Support for copies and renames
+
+We want to support copy tracing in a way that leaves it up to the commit backend
+to either record or detect copies. That should let us work with existing Git
+repos (Git does not record copies, it detects them on the fly) as well as with
+very large repos where detection would be too slow. See
+[design doc][copy-design-doc].
+
+## Forge integrations
+
+We would like to make it easier to work with various popular forges by providing
+something like `jj github submit`, `jj gitlab submit`, and `jj gerrit send`.
+
+## Better Rust API for UIs
+
+UIs like [gg] currently have to duplicate quite
+a bit of logic from `jj-cli`. We need to make this code not specific to the CLI
+(e.g. return status objects instead of printing messages) and move it into
+`jj-lib`.
+
+## RPC API
+
+One problem with writing tools using the Rust API is that they will only work
+with the backends they were compiled with. For example, a regular [gg] build
+will not work on Google repos because it doesn't have the backends necessary to
+load them. We want to provide an RPC API for tools that want to work with an
+unknown build of `jj` by having the tool run something like `jj api` to give it
+an address to talk to.
+
+In addition to helping with the problem of unknown backends, having an RPC API
+should make it easier for tools like VS Code that are not written in Rust. The
+RPC API will probably be at a higher abstraction level than the Rust API.
+
+See [design doc][api-design-doc].
+
+## Open-source cloud-based repos (server and daemon process)
+
+Google has an internal Jujutsu server backed by a database. This server allows
+commits and repos (operation logs) to be stored in the cloud (i.e. the database).
+Working copies can still be stored locally.
```

Comment:
While this part implies _some_ kind of object storage independent from Git, it isn't explicitly written as that. Any reason why? As I think having a BLAKE3 derived storage will help the later goals in the Roadmap.

Reply:
I'm not sure I understand your comment. I would prefer to not decide now what hash algorithm we're going to use, if that's what you're saying. Or are you saying that I should explicitly say that the server would use a hashing scheme that is *not* compatible with Git? It might actually be nice if we can have an option to be compatible with Git's hashing scheme so you can import a Git repo into our server.

Reply:
Maintaining compatibility with existing Git repositories would require allowing SHA‐1 into the system, which is going to be a can of worms. That’s not to say it might be the right trade‐off (especially since Git actually uses a hacked SHA‐1 that dances around the known easy collisions), but you’re sacrificing most of your potential guarantees right off the bat. (I do agree that committing to a hash function now makes no sense, but I don’t think that @PhilipMetzger meant that. (But I also agree with @PhilipMetzger that BLAKE3 is clearly the correct choice!))

Reply:
> I'm not sure I understand your comment.

Sorry for that.

> Or are you saying that I should explicitly say that the server would use a hashing scheme that is _not_ compatible with Git?

The intention was to get an answer if we could move on from Git as a commit storage model (it's our object store) and potentially build our own based on a BLAKE3 or even better hash in the future. This thought popped into my head while reading it, since Sapling is transitioning to BLAKE3 to unify the commit and CAS lookups, which probably is a major internal benefit to them. 

>  It might actually be nice if we can have an option to be compatible with Git's hashing scheme so you can import a Git repo into our server.

Ack, this makes definitely sense since a goal is to provide a seamless transition. But that shouldn't mean we'd need to support everything from Git.

Reply:
> The intention was to get an answer if we could move on from Git as a commit storage model

Yes, that has always been a goal. The local backend uses a different hashing scheme, for example. So does the Google backend. I'll close this but let me know if you think I should change anything here.